### PR TITLE
Akash scrum27and28

### PR DESCRIPTION
**Pull request by akashmisra1**
Added features to allow users to login and logout and have an account based on their google email.

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
 import { Button, Switch } from "@nextui-org/react";
 import { FaCircleInfo } from "react-icons/fa6";
 import RunVocoderForm from "./ui/run-vocoder-form";
+import { initFbase } from "../../firebase/firebaseApp";
+import { getAuth, signInWithPopup, GoogleAuthProvider } from "firebase/auth";
+import { useAuthState } from "react-firebase-hooks/auth";
 
 export default function Home() {
+
+  initFbase();
```
Comment (andrewpeng02) [inline src/app/page.tsx]: What is the point of this function if it just returns app? Can just import app directly if needed

File: src/app/page.tsx [hunk 2]
```diff
@@ -12,6 +27,8 @@ export default function Home() {
           <Button className="bg-blue-700 text-slate-300">File</Button>
         </div>
         <div className="flex">
+          <Button onClick={signIn} className="bg-blue-700 text-slate-300">Sign In</Button>
```
Comment (andrewpeng02) [inline src/app/page.tsx]: Add padding between the buttons so they don't appear squished
image

File: src/app/page.tsx [hunk 3]
```diff
@@ -12,6 +27,8 @@ export default function Home() {
...
+          <Button onClick={() => auth.signOut()} className="bg-blue-700 text-slate-300">Sign Out</Button>
```
Comment (andrewpeng02) [inline src/app/page.tsx]: If user is signed in, show sign out button; else, show sign in button

**Review by andrewpeng02 — CHANGES_REQUESTED** [review]

**Comment by akashmisra1** [comment]
Changes made @andrewpeng02 please check and lmk if it is ready to merge

File: src/app/page.tsx [hunk 4]
```diff
@@ -3,13 +3,13 @@
...
 import RunVocoderForm from "./ui/run-vocoder-form";
-import { initFbase } from "../../firebase/firebaseApp";
 import { getAuth, signInWithPopup, GoogleAuthProvider } from "firebase/auth";
 import { useAuthState } from "react-firebase-hooks/auth";
+import { app } from "../../firebase/firebaseApp";
 
 export default function Home() {
 
-  initFbase();
+  app;
```
Comment (andrewpeng02) [inline src/app/page.tsx:12]: remove line

File: src/app/page.tsx [hunk 5]
```diff
@@ -3,13 +3,13 @@
 import { Button, Switch } from "@nextui-org/react";
 import { FaCircleInfo } from "react-icons/fa6";
 import RunVocoderForm from "./ui/run-vocoder-form";
-import { initFbase } from "../../firebase/firebaseApp";
 import { getAuth, signInWithPopup, GoogleAuthProvider } from "firebase/auth";
 import { useAuthState } from "react-firebase-hooks/auth";
+import { app } from "../../firebase/firebaseApp";
 
 export default function Home() {
```
Comment (andrewpeng02) [inline src/app/page.tsx:11]: remove line

**Review by andrewpeng02 — APPROVED** [review]
after resolving comments, feel free to merge

**Comment by akashmisra1** [comment]
Need to keep those two lines in to initialize firebase on the website. Run into error otherwise.

**Comment by akashmisra1** [comment]
Merging changes with main branch

**Pull request merged**